common/testflinger_common/enums.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2024 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Job State and Test Phase Enums
+"""
+
+from strenum import StrEnum
```

Comment:
I haven't used `strenum` before but it might be worth checking if the behaviour of `strenum.StrEnum` is compatible with the `enum.StrEnum` available in Python 3.11 (in case the latter eventually replaces the former in our code). 

Even better, I wonder if it makes sense to derive our own `StrEnum` following the instructions [here](https://docs.python.org/3.8/library/enum.html#others). I don't think the definitions of `JobState` and `TestPhase` would change at all but we would avoid the dependency.

Reply:
I was looking at this too and I think there was some breaking change when using class(str, Enum) 
https://blog.pecar.me/python-enum

Reply:
Yeah, good StrEnum support seems to only start around python 3.11 it seems. From what I gather, using the mixin is only possible after 3.10, and there are also [compatibility problems](https://github.com/python/cpython/issues/100458) with that, and with how it works >=3.11.
This strenum module seems to be the most recommended option I could find, and is supposed to still work even in later versions so that we could easily transition from it if there are any differences from using enum.StrEnum. It does mention that it's not a perfect "drop-in" replacement, but it sounds like the biggest difference is that enum.StrEnum does a casefold() to lower case when you use auto() (which I'm not using). So I suspect we'll be able to migrate pretty easily. Just to test this, I also tried switching it for enum.StrEnum and at least in my current python (3.12.3) it seems to work.